tests/integration/components/async-button-test.js

```diff
@@ -0,0 +1,105 @@
+import Ember from 'ember';
+import { moduleForComponent, test } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+import wait from 'ember-test-helpers/wait';
+
+moduleForComponent('async-button', 'Integration | Component | async button', {
+  integration: true
+});
+
+test('it responds to a fulfilled closure promise', function(assert) {
+  let done = assert.async();
+
+  assert.expect(2);
+
+  this.set('default', 'Save');
```

Comment:
We prefer `set(this, 'default', 'Save')` as opposed to calling it off the object. Also, this could probably be:

```js
setProperties(this, {default: 'Save', pending: 'Saving', fulfilled: 'Saved'});
```